server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -1415,6 +1415,19 @@ private static boolean isDeprecatedParameter(String propName, IndexVersion index
         }
     }
 
+    public abstract static class DimensionBuilder extends Builder {
+
+        protected boolean isDimension = false;
+
+        public DimensionBuilder(String name) {
+            super(name);
+        }
+
+        void setDimension() {
```

Comment:
Hm that would be misleading, we want to mark that the mapper to be built will actually be a dimension, not that the builder is enabled to build a dimension (it is, de factor).

I really don't like the name though. This class creates objects that *can* be dimensions but are not necessarily. Any suggestions?

Reply:
It would be nice if we can move `Parameter<Boolean> dimension` into this builder, but that looks like a bigger refactoring to me. 

So passthrough field mapper invokes `setDimension()` here and then other field mappers (keyword, number, etc.) can become dimensions.  Maybe this method should be called `inheritDimension()`?

Reply:
Alright, switched to `inheritDimensionParameterFromParentObject`, more verbose for sure.